src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,399 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
+} from '@angular/core';
+import {
+  TreeViewColorTheme,
+  TreeViewConfiguration,
+  TreeViewEvent,
+  TreeViewEventType,
+  TreeNodeData,
+  TreeNodeDataRetriever,
+  PagedTreeNodeDataRetriever,
+  TreeNode,
+  FuiTreeViewComponentStyles,
+  NonRootTreeNode,
+} from './interfaces';
+import { FuiVirtualScrollerComponent } from '../virtual-scroller/virtual-scroller';
+import { Observable, fromEvent, Subscription } from 'rxjs';
+import { throttleTime } from 'rxjs/internal/operators';
+import { FuiTreeNodeComponent } from './tree-node-component';
+
+@Component({
+  selector: 'fui-tree-view',
+  template: `
+    <div id="fui-tree-view-component" [ngStyle]="treeViewStyles" [ngClass]="colorTheme">
+      <fui-virtual-scroller
+        [hidden]="loading || error"
+        #scroll
+        class="fui-virtual-scroller"
+        [items]="scrollViewArray"
+        [bufferAmount]="BUFFER_AMOUNT"
+      >
+        <fui-tree-node
+          *ngFor="let node of scroll.viewPortItems"
+          [node]="node"
+          [theme]="colorTheme"
+          [borders]="hasBorders"
+          [dataRetriever]="dataRetriever"
+          (onNodeEvent)="nodeEvent($event)"
+        ></fui-tree-node>
+      </fui-virtual-scroller>
+      <clr-icon *ngIf="loading" class="fui-loader fui-loader-animation" shape="fui-spinner"></clr-icon>
+      <div *ngIf="error" class="error">
+        <clr-icon class="fui-error-icon" shape="fui-error" aria-hidden="true"></clr-icon>
+        <p>Couldn't load content</p>
+      </div>
+    </div>
+  `,
+  styles: [
+    `
+      #fui-tree-view-component {
```

Comment:
Same here, please do not use the style directly in the angular component. Add it to ferui-design